packages/core-data/src/queried-data/reducer.js

```diff
@@ -32,6 +32,7 @@ import getQueryParts from './get-query-parts';
  * @return {number[]} Merged array of item IDs.
  */
 export function getMergedItemIds( itemIds, nextItemIds, page, perPage ) {
+	const receivedAllIds = page === 1 && perPage === -1;
```

Comment:
Isn't it the case that the entire function can be short-circuited if this is the case?

```suggestion
	const receivedAllIds = page === 1 && perPage === -1;
	if ( receivedAllIds ) {
		return nextItemIds;
	}
```

Reply:
It is! In fact that's also how my initial attempt looked like. It's just the result is would no longer be "sparse-like" as it normally is - it works for me if it works for you :-) I just updated the PR accordingly.

Reply:
Actually I'm wrong, the result will be exactly the same, maybe except for the fact that now it returns the original object instead of a new one.

Reply:
I just updated it to return a copy instead of the original list, just to maintain consistency with the other code branch.

Reply:
Personally I don't think it matters too much if it returns a copy, since the function doesn't really assert one way or the other, and in most cases throughout the code, we encourage to avoid object mutation and make no guarantees about behaviors (or misbehaviors) when mutation is performed.